.buildkite/pipeline.yml

```diff
@@ -49,3 +49,7 @@ steps:
       - *publish-android-artifacts-docker-container
     command: |
         .buildkite/publish-react-native-bridge-android-artifacts.sh
+
+  - label: "Run Danger CI"
+    command: |
+        yarn danger ci
```

Comment:
Good point - I believe `npx danger ci` would work here, and I've updated the code to that. Using a script in package.json is also an option if not. Under your current understanding, do you believe we would need to associate this command with a token (similar to how it has been [suggested](https://github.com/WordPress/gutenberg/pull/49547/files) in Gutenberg)?

Reply:
> Under your current understanding, do you believe we would need to associate this command with a token (similar to how it has been [suggested](https://github.com/WordPress/gutenberg/pull/49547/files) in Gutenberg)?

I interpret this to mean: do we need to add the to this `.buildkite/pipeline.yml` file? 

```yml
env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

In researching to understand your question and then answer it. I discovered a few things that invalidate some of the statements I shared elsewhere regards to tokens. 

First, to answer your question, **yes, I believe the token needs to be made available to Danger in order for it to post a comment on a PR.** 

However, part of my rationale for placing this in Gutenberg Mobile was to postpone a need to request a token for the Gutenberg repository. I have now come to understand that `secrets.GITHUB_TOKEN` is an [auto-created token](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) for all GitHub repositories using workflows. I came to understand this from reading the "Token Setup" of the "GitHub" tab within the [Setting up Danger to run on your CI](https://danger.systems/js/guides/getting_started.html#setting-up-danger-to-run-on-your-ci) section of the Danger documentation. 

Additionally, the `GITHUB_TOKEN` is [already used](https://github.com/WordPress/gutenberg/blob/721cb342060a5f6ddf34f25e5379e6d8a4c806ae/.github/workflows/build-plugin-zip.yml#L281-L282) in the Gutenberg project. So, it appears the `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` should "just work" in the Gutenberg repository. 

For Gutenberg Mobile, it may also "just work." However, there are two difference I notice in this repo:

1. The task is a Buildkite task — will using the token configuration line work? (We could just make a [new GitHub Workflow](https://github.com/wordpress-mobile/gutenberg-mobile/tree/trunk/.github/workflows).)
2. I see no existing use of `GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}` in this repo, so is that token available? 

All that said, it makes sense to me that we try to enable the CI integration in this repository to see if it succeeds. That may require merging an initial implementation. The configuration may need to be in `trunk` before they result in workflow runs.

Reply:
Thanks for this info. In reference to previous discussion on Buildkite reports [`npx: command not found`](https://buildkite.com/automattic/gutenberg-mobile/builds/5216#01874e63-0624-4770-b8f2-942a8663d57b), so that command must not be available. The Buildkite pipline file does use `npm` and shell scripts.

> I have now come to understand that secrets.GITHUB_TOKEN is an [auto-created token](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) for all GitHub repositories using workflows.
I was not aware of this, and this helps to clarify how the token should be used as I was uncertain where it actually came from or how it was generated. 

Looking at the Reassure docs, it appears they were updated very recently (5 hours ago) with a [new CI initialization command](https://github.com/callstack/reassure/commit/3f598bc0a6895003293ce260051f02993d184594) and further information on CI integration. According to Reassure's example [CI setup](https://github.com/callstack/reassure#ci-setup), the Dangerfile and `reassure-tests.sh` shell script live in the same directory. It is unclear to me if reporting test results in Gutenberg Mobile would actually work with the [Reassure library declaration](https://github.com/WordPress/gutenberg/pull/49221/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) in Gutenberg's package.json, Reassure tests, and [`reassure-tests.sh`](https://github.com/WordPress/gutenberg/pull/49547/files#diff-09a69d84b23588e21b7b3e9773e4f17254de23d564df2e28bcfbac7617d28314) script living in Gutenberg. At any rate, we may want to revise this approach here to more closely match Reassure's examples with a GitHub Workflow and token, as you've described.

Reply:
> [...] I was uncertain where [the token] actually came from or how it was generated.

In general, secrets are created in the GitHub repository settings. One can then utilize them as environment variables for CI tasks. However, `GITHUB_TOKEN` in particular appears to be a special token that is automatically created.